force-app/main/default/lwc/rd2ElevateCreditCardForm/rd2ElevateCreditCardForm.js

```diff
@@ -407,22 +453,44 @@ export default class rd2ElevateCreditCardForm extends LightningElement {
     }
 
     /**
-     * @description Concatenate the cardholder first and last names if there are any
-     * @returns A concatenated Cardholder Name string from the First and Last Name fields
+     * Handles changes made to fields in the form
+     * @param event Event containing field change information
      */
-    getCardholderName() {
-        const nameField = this.template.querySelector('[data-id="cardholderName"]');
-        return nameField.value;
+    handleFieldChange(event) {
+        const fieldName = event.target.name;
+        const newVal = event.detail.value;
+        this[fieldName] = newVal;
     }
 
     /**
-     * @description Concatenate the cardholder first and last names if there are any
-     * @returns A concatenated Cardholder Name string from the First and Last Name fields
+     * Handles population of Credit Card Billing Address fields based on the current donor
      */
-    @api
-    setCardholderName(donorName) {
-        const nameField = this.template.querySelector('[data-id="cardholderName"]');
-        nameField.value = donorName;
+    handleDefaultAddressButton() {
+        const donorId = this.rd2State.donorType === CONTACT_DONOR_TYPE
+            ? this.rd2State.contactId
+            : this.rd2State.accountId;
+        
+        if(!donorId) {
+            return;
+        }
+
+        this.searchingForAddress = true;
+        getDonorAddress({donorId: donorId})
+            .then((result) => {
+                if(result){
+                    this.billingCity= result.MailingCity__c;
+                    this.billingCountry = result.MailingCountry__c;
+                    this.billingPostalCode = result.MailingPostalCode__c;
+                    this.billingState = result.MailingState__c;
+                    this.billingStreet1 = result.MailingStreet__c;
+                    this.billingStreet2 = result.MailingStreet2__c;
```

Comment:
These fields will be namespaced in the package, can we import the field references and use that to grab the values out of the result?

Reply:
Excellent call out, I'm usually more careful about field references, on it!